### PR TITLE
Improved Documentation For Learning Rate Schedulers

### PR DESCRIPTION
**Pull request by gabeorlanski**
Partially addresses #4835.

Changes proposed in this pull request:

- Improved documentation and a basic config snippet example for the learning rate schedulers
- Moved the Pytorch wrapped learning rates into their own file `pytorch_lr_schedulers.py` so that they have their own docs page
- Added links to the original PyTorch LRSchedulers so that people can read their docs as well. 
- Added formula for Noam to its docs page.
- The only LR scheduler without an example is `slanted_triangular.py` as I have not used it before and do not know of a valid config for it.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [x] If this PR fixes a bug, I've added a test that will fail without my fix.
- [x] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [x] All GitHub Actions jobs for my pull request have passed.
- [x] **`codecov/patch`** reports high test coverage (at least 90%).
    You can find this under the "Actions" tab of the pull request once the other checks have finished.

### CODE REVIEW

**Review by AkshitaB — APPROVED** [review]
This is excellent! Thank you for the contribution.
